sympy/parsing/c/c_parser.py

```diff
@@ -233,40 +235,78 @@ def transform_var_decl(self, node):
                     child = next(children)
 
                 val = self.transform(child)
-                # List in case of variable assignment, FunctionCall node in case of a funcion call
+
+                # List in case of variable assignment,
+                # FunctionCall node in case of a function call
                 if (child.kind == cin.CursorKind.INTEGER_LITERAL
                     or child.kind == cin.CursorKind.UNEXPOSED_EXPR):
                     if (node.type.kind == cin.TypeKind.INT):
                         type = IntBaseType(String('integer'))
-                        value = Integer(val)
+                        # when only one unexposed_expr is assigned
+                        # e.g., int b = a;
+                        if isinstance(val, str):
+                            value = Symbol(val)
+                        # when val is integer
+                        # e.g., int b = 1;
+                        elif (isinstance(val, Integer)
+                            or isinstance(val, int)
+                            or isinstance(val, Float)
+                            or isinstance(val, float)):
+                            value = Integer(val)
+                        # when val is combination of both of the above
+                        # but in total only two nodes on rhs
+                        # e.g., int b = a * 1;
+                        else:
+                            value = val
+
                     elif (node.type.kind == cin.TypeKind.FLOAT):
                         type = FloatBaseType(String('real'))
-                        value = Float(val)
+                        # when only one unexposed_expr is assigned
```

Comment:
You don't need to repeat the comments. 
Once should be enough
You can just leave the eg if you want

Reply:
Ok